src/coreComponents/meshUtilities/PerforationData.cpp

```diff
@@ -151,15 +150,21 @@ void PerforationData::computeWellTransmissibility( MeshLevel const & mesh,
     real64 k1 = 0;
     real64 k2 = 0;
 
-    // compute the vector perforation - well elem center
-    // this vector will be used to decide whether this is a vectical well or not
-    localIndex const wellElemIndex   = m_wellElementIndex[iperf];
-    real64 vecWellElemCenterToPerf[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( wellElemCenter[wellElemIndex] );
-    LvArray::tensorOps::subtract< 3 >( vecWellElemCenterToPerf, m_location[iperf] );
+    // get the index of the well element, and of the neighboring well nodes
+    localIndex const wellElemIndex = m_wellElementIndex[iperf];
+    localIndex const topNode = elemToNodeMap[wellElemIndex][InternalWellGenerator::NodeLocation::TOP];
+    localIndex const bottomNode = elemToNodeMap[wellElemIndex][InternalWellGenerator::NodeLocation::BOTTOM];
```

Comment:
One funny test could be to swap the `0/1` values of `TOP` and `BOTTOM` to see if it still works. 😉

Reply:
I swapped the 0 and 1 (locally, not in the version I committed) and it worked. The integrated tests pass, except for the node list of the `WellElementSubRegion` that is obsviously different if 0 and 1 are swapped.

As a side note, I noticed that I broke an integrated test `deadoil_3ph_staircase_3d` with #901, I will fix it just after this PR using the Dead-Oil branch.